File: LeetCode-October-Challenge/ContainsDuplicate2.js
// var containsNearbyDuplicate = function(nums, k) {
//     let mp = new Map();
//     for(let i = 0; i < nums.length; ++i){
//         if(i - mp.get(nums[i]) <= k) return true;
//         mp.set(nums[i], i);
//     }
//     return false;
// };
var containsNearbyDuplicate = function(nums, k) {
    let mp = {};
    for(let i = 0; i < nums.length; ++i){
        if(i - mp[nums[i]] <= k) return true;
        mp[nums[i]] = i;
    }
    return false;
};
let nums = [1,0,1,1], k = 1;
console.log(containsNearbyDuplicate(nums, k));

File: LeetCode-October-Challenge/ContainsDuplicate2.java
import java.util.*;
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> mp = new TreeMap<>();
        for(int i = 0; i < nums.length; ++i){
            if(mp.containsKey(nums[i]) && i  -  mp.get(nums[i]) <= k) return true;
            mp.put(nums[i], i);
        }
        return false;
    }
}
public class ContainsDuplicate2 {
    public static void main(String[] args) {
        int[] nums = {1,0,1,1};
        int k = 1;
        Solution s = new Solution();
        System.out.println(s.containsNearbyDuplicate(nums, k));
    }
}
